packages/amplify-auth-hooks/src/use-sign-up.tsx

```diff
@@ -40,15 +40,15 @@ export const useSignUp = (): ((
     }
 
     const signupInfo = {
-      username: email,
-      password: password,
-      attributes: {},
+      username,
+      password,
+      attributes,
       validationData: validationDataArray,
     };
 
     try {
       const data = await Auth.signUp(signupInfo);
-      handleStateChange('confirmSignUp', data.user.getUsername());
+      handleStateChange('confirmSignUp', { username: data.user.getUsername() });
```

Comment:
FYI: This was a bug I found this code verification. I can break this out separately if needed.

To reproduce:
* Setup cognito with `DefaultEmailOption` set to `CONFIRM_WITH_CODE`
* Sign up via the material ui
* Check email (or phone) for code
* Enter code

**Expected:**
User is taken to the login page to sign in (I think this is intentional vs auto sign-in)

**Actual:**
Error: "Username is required" (I'm paraphrasing)

Based on `use-sign-in.tsx` in line 55, the object being passed in here must be `{ username }` format (object rather than a string).

Reply:
That's okay